include/libsemigroups/forest.hpp

```diff
@@ -93,59 +89,49 @@ namespace libsemigroups {
 
     ~Forest();
 
-    //! Add nodes to the Forest.
+    //! \brief Compare Forest objects for equality.
     //!
-    //! \param n the number of nodes to add.
+    //! This function returns \c true if and only if the parent and label of
+    //! every node in \c this coincides with the parent and label in \p that.
     //!
-    //! \returns
-    //! (None)
+    //! \param that the Forest for comparison.
     //!
-    //! \exceptions
-    //! \no_libsemigroups_except
-    //! \strong_guarantee
+    //! \returns whether or not the Forest objects are equal.
+    [[nodiscard]] bool operator==(Forest const& that) const {
+      return _parent == that._parent && _edge_label == that._edge_label;
+    }
+
+    //! \brief Compare Forest objects for inequality.
     //!
-    //! \complexity
-    //! At most linear in `number_of_nodes() + n`.
     //!
-    //! \iterator_validity
-    //! \iterator_invalid
-    // TODO to cpp
-    void add_nodes(size_t n) {
-      size_t const old_nr_nodes = number_of_nodes();
-      try {
-        _edge_label.insert(
-            _edge_label.cend(), n, static_cast<size_t>(UNDEFINED));
-        _parent.insert(_parent.cend(), n, static_cast<size_t>(UNDEFINED));
-      } catch (...) {
-        _edge_label.resize(old_nr_nodes);
-        _parent.resize(old_nr_nodes);
-        throw;
-      }
+    //! \param that the Forest for comparison.
+    //!
+    //! \returns whether or not the Forest objects are not equal.
```

Comment:
Underneath `\brief Compare Forest objects for inequality`, should we have something like 
```
This function returns the negation of `Operator==(that)`
```

The fact that `operator!=` jumps straight into the parameters whereas `operator==` doesn't feels a bit inconsistent. Not sure about this one though